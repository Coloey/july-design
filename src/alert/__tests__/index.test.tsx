import React from 'react';
import { fireEvent, render,screen} from '@testing-library/react';
import Alert from '../index';

describe('<Alert />', () => {
  it('should render default', () => {
    const { container } = render(<Alert>hello world</Alert>);
    expect(container).toMatchSnapshot();
  });
  
  it("render with type and message",() => {
    const types:any[]= ['success','info','error','warning']  
    types.forEach((t) => {
      render(<Alert type={t} message={t}></Alert>)
      expect(screen.getByText(t)).toMatchSnapshot();
    })    
  })
})

describe('Alert[onClose]',() => {
  it('onClick',() => {
    const onClose = jest.fn()
    render(<Alert closable onClose={onClose}></Alert>)
    fireEvent.click(screen.getByText("x"))
    expect(onClose).toBeCalled();
    expect(onClose).toBeCalledTimes(1)
  }) ;

})
