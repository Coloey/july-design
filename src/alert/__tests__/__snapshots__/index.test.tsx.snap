// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> render with type and message 1`] = `
<div
  class="alertMes"
>
  success
</div>
`;

exports[`<Alert /> render with type and message 2`] = `
<div
  class="alertMes"
>
  info
</div>
`;

exports[`<Alert /> render with type and message 3`] = `
<div
  class="alertMes"
>
  error
</div>
`;

exports[`<Alert /> render with type and message 4`] = `
<div
  class="alertMes"
>
  warning
</div>
`;

exports[`<Alert /> should render default 1`] = `
<div>
  <div
    class="alert info"
    style="opacity: 1;"
  >
    <div
      class="alertMes"
    />
    hello world
  </div>
</div>
`;
